framework/web/Response.php

```diff
@@ -857,18 +857,22 @@ public function redirect($url, $statusCode = 302, $checkAjax = true)
             // ensure the route is absolute
             $url[0] = '/' . ltrim($url[0], '/');
         }
+        $request = Yii::$app->getRequest();
         $url = Url::to($url);
         if (strncmp($url, '/', 1) === 0 && strncmp($url, '//', 2) !== 0) {
-            $url = Yii::$app->getRequest()->getHostInfo() . $url;
+            $url = $request->getHostInfo() . $url;
         }
 
         if ($checkAjax) {
-            if (Yii::$app->getRequest()->getIsAjax()) {
-                if (Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null && $statusCode === 302) {
+            if ($request->getIsAjax()) {
+                $is302 = $statusCode === 302;
+                if ($is302 && $request->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null) {
                     // Ajax 302 redirect in IE does not work. Change status code to 200. See https://github.com/yiisoft/yii2/issues/9670
                     $statusCode = 200;
+                } elseif ($is302 && $request->isAjax && $request->isPost && preg_match('/Trident.*\brv\:11\./' /* IE11 */, $request->userAgent)) {
+                    $statusCode = 308;
```

Comment:
Would that work in case `X-Ie-Redirect-Compatibility` present?

Reply:
Good idea, I'll test it more.

Reply:
Notes for me:
- https://github.com/yiisoft/jquery-pjax/issues/37
- https://github.com/yiisoft/jquery-pjax/releases/tag/v2.0.6
- https://github.com/yiisoft/jquery-pjax/pull/38

Reply:
As I look at the code the two are the same. `X-Ie-Redirect-Compatibility` header is restricted to only pjax.
I'll think better about it and test it, but probably the `X-Ie-Redirect-Compatibility` header header condition is not needed.

Reply:
| check list for redirects | result | 
| ------------ |----|
| pjax         |     |
| pjax + POST + IE 11 |    |
| pjax + GET + IE 11   |    |
| pjax + POST + not IE 11 |    |
| pjax + GET + not IE 11 |    |
| ajax |     |
| ajax + POST + IE 11 |    |
| ajax + GET + IE 11   |    |
| ajax + POST + not IE 11 |    |
| ajax + GET + not IE 11 |    |

Reply:
| check list for http codes | ie11 result |
|---|---|  
|301| |
|302| ✔️ |
|308| |

Reply:
@samdark reviewable / testable